--- upstart-0.5.0/init/job_process.c.org	2008-11-12 08:56:07.156866362 +0100
+++ upstart-0.5.0/init/job_process.c	2008-11-12 08:56:10.980199014 +0100
@@ -507,16 +507,11 @@
 	snprintf (filename, sizeof (filename), "/proc/%d/oom_adj", getpid ());
 
 	fd = fopen (filename, "w");
-	if (! fd) {
-		nih_error_raise_system ();
-		job_process_error_abort (fds[1], JOB_PROCESS_ERROR_OOM_ADJ, 0);
-	}
-
-	fprintf (fd, "%d\n", class->oom_adj);
-
-	if (fclose (fd)) {
-		nih_error_raise_system ();
-		job_process_error_abort (fds[1], JOB_PROCESS_ERROR_OOM_ADJ, 0);
+	if (fd) {
+		fprintf (fd, "%d\n", class->oom_adj);
+		fclose (fd);
+	} else {
+		nih_warn (_("Failed to open: %s"), filename);
 	}
 
 	/* Change the root directory, confining path resolution within it;
