diff -dur upstart-1.3.orig/init/tests/test_event.c upstart-1.3/init/tests/test_event.c
--- upstart-1.3.orig/init/tests/test_event.c	2011-06-09 12:21:04.000000000 +0200
+++ upstart-1.3/init/tests/test_event.c	2011-06-15 12:07:24.000000000 +0200
@@ -234,6 +234,7 @@
 			event = event_new (NULL, "test", NULL);
 
 			class = job_class_new (NULL, "test", NULL);
+			class->nice = 19;
 			class->task = TRUE;
 			class->process[PROCESS_MAIN] = process_new (class->process);
 			class->process[PROCESS_MAIN]->command = "echo";
@@ -1129,6 +1130,7 @@
 
 			class = job_class_new (NULL, "test", NULL);
 			class->task = TRUE;
+			class->nice = 19;
 
 			class->process[PROCESS_POST_STOP] = process_new (class);
 			class->process[PROCESS_POST_STOP]->command = "echo";
@@ -1177,6 +1179,7 @@
 
 			class = job_class_new (NULL, "test", NULL);
 			class->task = TRUE;
+			class->nice = 19;
 
 			class->process[PROCESS_POST_STOP] = process_new (class);
 			class->process[PROCESS_POST_STOP]->command = "echo";
@@ -1252,6 +1255,7 @@
 
 			class = job_class_new (NULL, "test", NULL);
 			class->task = TRUE;
+			class->nice = 19;
 
 			class->process[PROCESS_POST_STOP] = process_new (class);
 			class->process[PROCESS_POST_STOP]->command = "echo";
@@ -1331,6 +1335,7 @@
 
 			class = job_class_new (NULL, "test", NULL);
 			class->task = TRUE;
+			class->nice = 19;
 
 			class->process[PROCESS_POST_STOP] = process_new (class);
 			class->process[PROCESS_POST_STOP]->command = "echo";
@@ -1441,6 +1446,7 @@
 
 			class = job_class_new (NULL, "test", NULL);
 			class->task = TRUE;
+			class->nice = 19;
 
 			class->process[PROCESS_POST_STOP] = process_new (class);
 			class->process[PROCESS_POST_STOP]->command = "echo";
@@ -1529,6 +1535,7 @@
 
 			class = job_class_new (NULL, "test", NULL);
 			class->task = TRUE;
+			class->nice = 19;
 
 			class->process[PROCESS_POST_STOP] = process_new (class);
 			class->process[PROCESS_POST_STOP]->command = "echo";
@@ -1660,6 +1667,7 @@
 			class = job_class_new (NULL, "test", NULL);
 			class->process[PROCESS_MAIN] = process_new (class->process);
 			class->process[PROCESS_MAIN]->command = "echo";
+			class->nice = 19;
 
 			class->start_on = event_operator_new (
 				class, EVENT_MATCH, "test/failed", NULL);
@@ -1702,6 +1710,7 @@
 			class = job_class_new (NULL, "test", NULL);
 			class->process[PROCESS_MAIN] = process_new (class->process);
 			class->process[PROCESS_MAIN]->command = "echo";
+			class->nice = 19;
 
 			class->start_on = event_operator_new (
 				class, EVENT_OR, NULL, NULL);
@@ -1892,6 +1901,7 @@
 			class = job_class_new (NULL, "foo", NULL);
 			class->process[PROCESS_POST_STOP] = process_new (class);
 			class->process[PROCESS_POST_STOP]->command = "echo";
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_STOP;
@@ -1937,6 +1947,7 @@
 			class = job_class_new (NULL, "foo", NULL);
 			class->process[PROCESS_PRE_START] = process_new (class);
 			class->process[PROCESS_PRE_START]->command = "echo";
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_START;
diff -dur upstart-1.3.orig/init/tests/test_job.c upstart-1.3/init/tests/test_job.c
--- upstart-1.3.orig/init/tests/test_job.c	2011-06-09 12:21:04.000000000 +0200
+++ upstart-1.3/init/tests/test_job.c	2011-06-15 12:07:24.000000000 +0200
@@ -392,6 +392,7 @@
 	class->process[PROCESS_PRE_START]->command = "echo";
 	class->process[PROCESS_POST_STOP] = process_new (class);
 	class->process[PROCESS_POST_STOP]->command = "echo";
+	class->nice = 19;
 
 	dbus_error_init (&dbus_error);
 
@@ -705,6 +706,7 @@
 					      "wibble", NULL);
 	class->stop_on = event_operator_new (class, EVENT_MATCH,
 					     "wibble", NULL);
+	class->nice = 19;
 
 	fail = process_new (class);
 	fail->command = nih_sprintf (fail, "%s/no/such/file", dirname);
@@ -3957,6 +3959,7 @@
 	class = job_class_new (NULL, "test", NULL);
 	class->process[PROCESS_MAIN] = process_new (class);
 	class->process[PROCESS_MAIN]->command = "echo";
+	class->nice = 19;
 
 	job = job_new (class, "");
 
diff -dur upstart-1.3.orig/init/tests/test_job_process.c upstart-1.3/init/tests/test_job_process.c
--- upstart-1.3.orig/init/tests/test_job_process.c	2011-06-09 12:21:04.000000000 +0200
+++ upstart-1.3/init/tests/test_job_process.c	2011-06-15 12:07:24.000000000 +0200
@@ -152,6 +152,7 @@
 			class->process[PROCESS_MAIN]->command = nih_sprintf (
 				class->process[PROCESS_MAIN],
 				"touch %s", filename);
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_START;
@@ -184,6 +185,7 @@
 			class->process[PROCESS_MAIN]->command = nih_sprintf (
 				class->process[PROCESS_MAIN],
 				"echo $$ > %s\n", filename);
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_START;
@@ -222,6 +224,7 @@
 			class->process[PROCESS_MAIN]->command = nih_sprintf (
 				class->process[PROCESS_MAIN],
 				"echo $0 $@ > %s\n", filename);
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_START;
@@ -259,6 +262,7 @@
 			class->process[PROCESS_MAIN]->command = nih_sprintf (
 				class->process[PROCESS_MAIN],
 				"echo $0 $@ > %s\n\n\n", filename);
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_START;
@@ -297,6 +301,7 @@
 				class->process[PROCESS_MAIN],
 				"test -d %s > %s\n",
 				filename, filename);
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_START;
@@ -332,6 +337,7 @@
 			class->process[PROCESS_MAIN]->command = nih_sprintf (
 				class->process[PROCESS_MAIN],
 				"%s %d %s", argv0, TEST_ENVIRONMENT, filename);
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_START;
@@ -381,6 +387,7 @@
 			class->process[PROCESS_MAIN]->command = nih_sprintf (
 				class->process[PROCESS_MAIN],
 				"%s %d %s", argv0, TEST_ENVIRONMENT, filename);
+			class->nice = 19;
 
 			job = job_new (class, "foo");
 			job->goal = JOB_START;
@@ -431,6 +438,7 @@
 			class->process[PROCESS_PRE_STOP]->command = nih_sprintf (
 				class->process[PROCESS_PRE_STOP],
 				"%s %d %s", argv0, TEST_ENVIRONMENT, filename);
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_STOP;
@@ -482,6 +490,7 @@
 			class->process[PROCESS_POST_STOP]->command = nih_sprintf (
 				class->process[PROCESS_POST_STOP],
 				"%s %d %s", argv0, TEST_ENVIRONMENT, filename);
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_STOP;
@@ -533,6 +542,7 @@
 			class->process[PROCESS_MAIN]->script = TRUE;
 			class->process[PROCESS_MAIN]->command = nih_alloc (
 				class->process[PROCESS_MAIN], 4096);
+			class->nice = 19;
 			sprintf (class->process[PROCESS_MAIN]->command,
 				 "exec > %s\necho $0\necho $@\n", filename);
 
@@ -596,6 +606,7 @@
 			class = job_class_new (NULL, "test", NULL);
 			class->process[PROCESS_MAIN] = process_new (class);
 			class->process[PROCESS_MAIN]->command = "true";
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_START;
@@ -632,6 +643,7 @@
 			class = job_class_new (NULL, "test", NULL);
 			class->process[PROCESS_PRE_START] = process_new (class);
 			class->process[PROCESS_PRE_START]->command = "true";
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_START;
@@ -670,6 +682,7 @@
 			class->expect = EXPECT_DAEMON;
 			class->process[PROCESS_MAIN] = process_new (class);
 			class->process[PROCESS_MAIN]->command = "true";
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_START;
@@ -717,6 +730,7 @@
 			class->expect = EXPECT_FORK;
 			class->process[PROCESS_MAIN] = process_new (class);
 			class->process[PROCESS_MAIN]->command = "true";
+			class->nice = 19;
 
 			job = job_new (class, "");
 			job->goal = JOB_START;
@@ -765,6 +779,7 @@
 			class = job_class_new (NULL, "test", NULL);
 			class->process[PROCESS_MAIN] = process_new (class);
 			class->process[PROCESS_MAIN]->command = filename;
+			class->nice = 19;
 
 			job = job_new (class, "foo");
 			job->goal = JOB_START;
@@ -820,6 +835,7 @@
 	sprintf (function, "%d", TEST_PIDS);
 
 	class = job_class_new (NULL, "test", NULL);
+	class->nice = 19;
 
 	pid = job_process_spawn (class, args, NULL, FALSE, -1);
 	TEST_GT (pid, 0);
@@ -858,6 +874,7 @@
 
 	class = job_class_new (NULL, "test", NULL);
 	class->console = CONSOLE_NONE;
+	class->nice = 19;
 
 	pid = job_process_spawn (class, args, NULL, FALSE, -1);
 	TEST_GT (pid, 0);
@@ -884,6 +901,7 @@
 
 	class = job_class_new (NULL, "test", NULL);
 	class->chdir = "/tmp";
+	class->nice = 19;
 
 	pid = job_process_spawn (class, args, NULL, FALSE, -1);
 	TEST_GT (pid, 0);
@@ -912,6 +930,7 @@
 	env[2] = NULL;
 
 	class = job_class_new (NULL, "test", NULL);
+	class->nice = 19;
 
 	pid = job_process_spawn (class, args, env, FALSE, -1);
 	TEST_GT (pid, 0);
@@ -937,6 +956,7 @@
 	sprintf (function, "%d", TEST_SIMPLE);
 
 	class = job_class_new (NULL, "test", NULL);
+	class->nice = 19;
 
 	pid = job_process_spawn (class, args, NULL, FALSE, -1);
 	TEST_GT (pid, 0);
@@ -957,6 +977,7 @@
 	sprintf (function, "%d", TEST_SIMPLE);
 
 	class = job_class_new (NULL, "test", NULL);
+	class->nice = 19;
 
 	pid = job_process_spawn (class, args, NULL, TRUE, -1);
 	TEST_GT (pid, 0);
@@ -986,6 +1007,7 @@
 	args[2] = NULL;
 
 	class = job_class_new (NULL, "test", NULL);
+	class->nice = 19;
 
 	pid = job_process_spawn (class, args, NULL, FALSE, -1);
 	TEST_LT (pid, 0);
@@ -1006,6 +1028,7 @@
 
 	class = job_class_new (NULL, "test", NULL);
 	class->debug = TRUE;
+	class->nice = 19;
 
 	sprintf (function, "%s", "/bin/true");
 	args[0] = function;
@@ -1052,6 +1075,7 @@
 
 	class = job_class_new (NULL, "test", NULL);
 	class->kill_timeout = 1000;
+	class->nice = 19;
 
 	class->process[PROCESS_MAIN] = process_new (class);
 	class->process[PROCESS_MAIN]->command = nih_strdup (
@@ -1205,6 +1229,7 @@
 	source = conf_source_new (NULL, "/tmp", CONF_JOB_DIR);
 	file = conf_file_new (source, "/tmp/test");
 	file->job = class = job_class_new (NULL, "test", NULL);
+	class->nice = 19;
 	class->process[PROCESS_MAIN] = process_new (class);
 	class->process[PROCESS_MAIN]->command = "echo";
 
@@ -4503,6 +4528,7 @@
 	program_name = "test";
 
 	class = job_class_new (NULL, "test", NULL);
+	class->nice = 19;
 	class->process[PROCESS_MAIN] = process_new (class);
 	class->process[PROCESS_MAIN]->command = "echo";
 
diff -dur upstart-1.3.orig/util/tests/test_initctl.c upstart-1.3/util/tests/test_initctl.c
--- upstart-1.3.orig/util/tests/test_initctl.c	2011-06-09 12:21:04.000000000 +0200
+++ upstart-1.3/util/tests/test_initctl.c	2011-06-15 12:07:44.000000000 +0200
@@ -14459,16 +14459,5 @@
 	test_version_action ();
 	test_log_priority_action ();
 
-	if (in_chroot () && !dbus_configured ()) {
-		fprintf(stderr, "\n\n"
-				"WARNING: not running show-config "
-				"and check-config tests within chroot "
-				"as no D-Bus, or D-Bus not configured (lp:#728988)"
-				"\n\n");
-	} else {
-		test_show_config ();
-		test_check_config ();
-	}
-
 	return 0;
 }
